### PR TITLE
Added menu markdown, dartpads, and images

### PR DESCRIPTION
**Pull request by JoseAlba**
#### Don't forget:
- [x] Are you sure you mean this repo? This repository is a reference to the [main Flutter repository](https://github.com/flutter/flutter). Code-level issues should be created there.
- [ ] Link to GitHub issues it solves. ```closes #1234```
- [x] Sign the CLA bot. You can do this once the pull request is opened.

Be sure to read the [Flutter guide on contributing](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md).

### Related issues
 - Type `#` and the issue number or `#` and select the from the inline drop down.

### CODE REVIEW

File: docs/components/dartpad/menus/dropdown/main.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+}
+
+class DropdownMenuDemo extends StatefulWidget {
+  @override
+  _DropdownMenuDemoState createState() => _DropdownMenuDemoState();
+}
+
+class _DropdownMenuDemoState extends State<DropdownMenuDemo> {
+  Map<int, bool> countToValue = <int, bool>{};
```
Comment (rami-a) [inline docs/components/dartpad/menus/dropdown/main.dart:22]: This looks like it isn't being used, I think you can remove it and make this a statelesswidget instead

File: docs/components/dartpad/menus/theme/main.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  }
+}
+
+class DropdownMenuDemo extends StatefulWidget {
+  @override
+  _DropdownMenuDemoState createState() => _DropdownMenuDemoState();
+}
+
+class _DropdownMenuDemoState extends State<DropdownMenuDemo> {
+  Map<int, bool> countToValue = <int, bool>{};
```
Comment (rami-a) [inline docs/components/dartpad/menus/theme/main.dart:23]: Same here, remove this property and make it a statelesswidget

File: docs/components/menus.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+![Menu hero example showing two cascading menus](assets/menus/menus-hero.png)
+
+# Contents
+
+- [Using menus](#using-menus)
+- [Installing menus](#installing-menus)
+- [Making menus accessible](#making-menus-accessible)
+- [Dropdown menus](#dropdown-menus)
+- [Exposed dropdown menus](#exposed-dropdown-menus)
+- [Theming](#theming)
```
Comment (rami-a) [inline docs/components/menus.md:24]: I think the table of contents should represent only the top level headers

File: docs/components/menus.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+## Using menus
+
+A menu displays a list of choices on a temporary surface. They appear when users interact with a button, action, or other control.
+
+### Installing menus
```
Comment (rami-a) [inline docs/components/menus.md:30]: Remove this header since there's no content in it

File: docs/components/menus.md [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+1. Cascading menu indicator (optional)
+1. Selection state (optional)
+
+### Container attributes
+
+&nbsp; | Property
+------ | ---------
+**Color** | Use the `PopupMenuButton` widget. Within this widget use the `Colors` property
+**Height** | Use the `PopupMenuButton` widget. Within this widget use the `padding` property.
+**Overlay window** | Use the `PopupMenuButton` widget. Within this widget use `itemBuilder` property. Each item added to this list will add an additional window.
```
Comment (rami-a) [inline docs/components/menus.md:56]: This attribute doesn't seem correct. I'm not sure what is meant by OverlayWindow

File: docs/components/menus.md [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+    setState(() {
+      dropdownValue = value;
+    });
+  },
+),
+```
+
+## Theming
+
+Menus support [Material Theming](https://material.io/components/chips/#theming) and can be customized in terms of color, typography and shape.
```
Comment (rami-a) [inline docs/components/menus.md:200]: I think that material.io link is wrong, it refers to chips

File: docs/components/menus.md [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+
+### Theming example
+
+The following is an example of a dropdown menu:
+
+- [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/theme&gh_ref=develop)
```
Comment (rami-a) [inline docs/components/menus.md:206]:
```suggestion
- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/theme&gh_ref=develop)
```

**Review by rami-a — APPROVED** [review]
LGTM with a few comments and nits

File: docs/components/menus.md [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+
+#### Dropdown menu example
+
+The following is an example of a dropdown menu:
+
+- [Class definition](https://api.flutter.dev/flutter/material/PopupMenuButton-class.html)
+- [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/popup_menu.dart)
+- [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/dropdown&gh_ref=develop)
+
+![Dropdown menu example](assets/menus/dropdown_menu_example.png)
```
Comment (perclasson) [inline docs/components/menus.md:116]: Looks like the appbar color in the background is not the correct color, i.e. the material purple color. Is it correct in the demo?

File: docs/components/dartpad/menus/exposed_dropdown/main.dart [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Exposed Menu Demo',
+      debugShowCheckedModeBanner: false,
+      home: ExposedDropdownMenuDemo(),
```
Comment (perclasson) [inline docs/components/dartpad/menus/exposed_dropdown/main.dart:11]: I think we should set the `ThemeData.primarColor` here to be purple.

File: docs/components/menus.md [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+
+&nbsp;         | Property
+-------------- | ------------------------
+Adding cascading menu | Not currently supported.
+
+### Selection state attributes
+
+&nbsp;         | Property
+-------------- | ------------------------
+ | Use the `PopupMenuButton` widget. Within this widget use `itemBuilder`. Within the list of itemBuilder add a `CheckedPopupMenuItem`.
```
Comment (perclasson) [inline docs/components/menus.md:94]: Row here is missing the value of the first column, Or should it be empty? Doesn't look correct when I preview.

**Review by perclasson — APPROVED** [review]
LGTM

File: docs/components/dartpad/menus/dropdown/main.dart (hunk 1 above)
Comment (JoseAlba) [reply docs/components/dartpad/menus/dropdown/main.dart:22]: Done

File: docs/components/dartpad/menus/theme/main.dart (hunk 2 above)
Comment (JoseAlba) [reply docs/components/dartpad/menus/theme/main.dart:23]: Done

File: docs/components/menus.md (hunk 4 above)
Comment (JoseAlba) [reply docs/components/menus.md:30]: Done

File: docs/components/menus.md (hunk 6 above)
Comment (JoseAlba) [reply docs/components/menus.md:200]: Done

File: docs/components/menus.md (hunk 7 above)
Comment (JoseAlba) [reply docs/components/menus.md:206]: Done

File: docs/components/dartpad/menus/exposed_dropdown/main.dart (hunk 9 above)
Comment (JoseAlba) [reply docs/components/dartpad/menus/exposed_dropdown/main.dart:11]: I didn't do it for this example but I did it for the dropdown example. That way it matches the demo.

File: docs/components/menus.md (hunk 10 above)
Comment (JoseAlba) [reply docs/components/menus.md:94]: Done

File: docs/components/menus.md (hunk 8 above)
Comment (JoseAlba) [reply docs/components/menus.md:116]: Done

File: docs/components/menus.md (hunk 5 above)
Comment (JoseAlba) [reply docs/components/menus.md:56]: Done

File: docs/components/menus.md (hunk 3 above)
Comment (JoseAlba) [reply docs/components/menus.md:24]: Done

File: docs/components/menus.md [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+
+#### Exposed dropdown menu example
+
+The following is an example of a dropdown menu:
+
+- [Class definition](https://api.flutter.dev/flutter/material/DropdownButton-class.html)
+- [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/dropdown.dart)
+- [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/exposed_dropdown&gh_ref=develop)
+
+![Exposed Dropdown menu example](assets/menus/exposed_dropdown_menu_example.png)
```
Comment (mingjane-work) [inline docs/components/menus.md:154]: Please update alt-text to:

"Exposed dropdown menu example listing 6 options"

File: docs/components/menus.md [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+
+Menus support [Material Theming](https://material.io/components/chips/#theming) and can be customized in terms of color, typography and shape.
+
+### Theming example
+
+The following is an example of a dropdown menu:
+
+- [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/theme&gh_ref=develop)
+
+![Dropdown menu themed example](assets/menus/dropdown_menu_themed_example.png)
```
Comment (mingjane-work) [inline docs/components/menus.md:203]: Replace alt-text with:

"Dropdown menu with dark brown text and gray icons."

File: docs/components/menus.md (hunk 11 above)
Comment (JoseAlba) [reply docs/components/menus.md:154]: Done

File: docs/components/menus.md (hunk 12 above)
Comment (JoseAlba) [reply docs/components/menus.md:203]: Done

**Review by mingjane-work — APPROVED** [review]
LGTM

**Pull request merged**